c2corg_ui/static/js/documentediting.js

```diff
@@ -445,17 +445,21 @@ app.DocumentEditingController.prototype.hasMissingProps = function(doc, showErro
       hasError = (!doc['locales'] || !doc['locales'][0][field]);
     } else if (field === 'activities') {
       hasError = (!doc['activities'] || doc['activities'].length === 0);
-    } else if (field === 'routes') {
-      hasError = (!doc['associations'] || doc['associations']['routes'].length === 0);
+    } else if (field === 'routes' || field === 'waypoints') {
+      hasError = (!doc['associations'] || doc['associations'][field].length === 0);
     } else if (field === 'latitude' || field === 'longitude') {
       hasError = (!doc['lonlat'] || (doc['lonlat'][field] === null || doc['lonlat'][field] === undefined));
     } else if (field === 'date_start') {
       hasError = (doc['date_start'] === null || doc['date_start']  === undefined);
     } else {
       hasError = (!doc[field] || doc[field] === null || doc[field] === undefined);
     }
+
     if (hasError) {
       if (showError) {
+        if (field === 'routes' || field === 'waypoints') {
+          field = 'no associated ' + field;
```

Comment:
"no associated" is not possible because the order of the words changes according to the language.

See also https://github.com/c2corg/v6_ui/blob/master/c2corg_ui/templates/i18n.html